### PR TITLE
Filter by resources type using checkbox

### PR DESCRIPTION
**Pull request by ronitjadhav**
#13 
- Added a check box for each resource type
- Filter the resource according to the checkbox

image

### CODE REVIEW

File: qgis_hub_plugin/gui/resource_browser.py [hunk 1]
```diff
@@ -60,15 +71,27 @@ def __init__(self, parent=None, iface=None):
...
+            # total = response.get("total")
+            # previous_url = response.get("previous")
+            # next_url = response.get("next")
+            self.resources = response.get("results", {})
+
+        geopackage_checked = self.checkBoxGeopackage.isChecked()
+        style_checked = self.checkBoxStyle.isChecked()
+        model_checked = self.checkBoxModel.isChecked()
+
+        filtered_resources = [
```
Comment (ismailsunni) [inline qgis_hub_plugin/gui/resource_browser.py:85]: Actually, I prefer to use `qsortfilterproxymodel` for filtering (and search also sorting in the future). Could you try to use `qsortfilterproxymodel`? It's more challenging to use of course :)

File: qgis_hub_plugin/gui/resource_browser.py [hunk 2]
```diff
@@ -50,6 +51,16 @@ def __init__(self, parent=None, iface=None):
 
         self.pushButtonDownload.clicked.connect(self.download_resource)
 
+        self.checkBoxGeopackage.stateChanged.connect(
```
Comment (ismailsunni) [inline qgis_hub_plugin/gui/resource_browser.py:54]: If we check in the hub API, there are 5 resource types
image

It seems the API doesn't return all five resource types. Could you check it? Add the missing 2 check boxes after that.

Comment (ronitjadhav) [reply qgis_hub_plugin/gui/resource_browser.py:58]: I checked the API URL used, it only returns three types of resources.
Is there any other API I can use, or can you direct me to the place where I can find the API's.

File: qgis_hub_plugin/gui/resource_browser.py (hunk 1 above)
Comment (ronitjadhav) [reply qgis_hub_plugin/gui/resource_browser.py]: I have implemented the `qsortfilterproxymodel` for filtering.

**Review by ismailsunni — APPROVED** [review]
LGTM, thanks @ronitjadhav

File: qgis_hub_plugin/gui/resource_browser.py (hunk 2 above)
Comment (ismailsunni) [reply qgis_hub_plugin/gui/resource_browser.py:58]: It's not really documented, but it's from this https://github.com/qgis/QGIS-Django/blob/master/qgis-app/api/urls.py

Comment (ismailsunni) [reply qgis_hub_plugin/gui/resource_browser.py:58]: I will create an issue in that repo to include the other resource type also.

Comment (ismailsunni) [reply qgis_hub_plugin/gui/resource_browser.py:58]: https://github.com/qgis/QGIS-Django/issues/280

**Pull request merged**